src/v/raft/consensus.cc

```diff
@@ -1220,6 +1220,115 @@ consensus::abort_configuration_change(model::revision_id revision) {
     co_return errc::success;
 }
 
+template<typename Func>
+ss::future<std::error_code>
+consensus::force_replace_configuration_locally(Func&& f) {
+    return _op_lock.get_units()
+      .then([this, f = std::forward<Func>(f)](ssx::semaphore_units u) mutable {
+          // It is to be noted that we have no checks here that validate if
+          // there is a configuration change in progress. This is by design
+          // and must be used with caution.
+          if (unlikely(_leader_id)) {
+              // If there is an existing leader, we do not need to perform this
+              // unclean repair. This is intended for exceptional cases where a
+              // leader cannot be elected due to lack of majority.
+              vlog(
+                _ctxlog.debug,
+                "Rejecting force configuration update as there is an elected "
+                "leader: {}",
+                _leader_id.value());
+              return ss::make_ready_future<std::error_code>(
+                errc::invalid_configuration_update);
```

Comment:
I think this validation may lead to the situation in which force reconfiguraion will never finish (we are not validating leadership in `controller_backend`) imagine someone requesting a force reconfiguration and then partition elected a leader.